ykcompile/src/lib.rs

```diff
@@ -323,14 +318,17 @@ impl<TT> TraceCompiler<TT> {
                             }
                             _ => unreachable!(),
                         };
-                        if !store {
+                        let nextloc = if store {
```

Comment:
I think we can just do:
```
self.free_if_temp(curloc);
curloc = if store {
    Location::Add(temp);
} else {
    ...
    Location::Register(temp);
}
```

Reply:
But you have to free `curloc`.

Reply:
Doing it this way we don't duplicate the freeing.

Reply:
I don't follow. I am freeing `curloc` and then I'm overwriting it with a new location.